src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ImagePreviewer.java

```diff
@@ -522,6 +553,29 @@ private static void onPreviewImage(DocDisplay display,
       panel.setPopupPosition(coordinates.getPageX(), coordinates.getPageY() + 20);
       panel.show();
    }
+   
+   private static final native void injectAttributes(Element el, String attributes)
+   /*-{
+      // create an element, inject inner HTML, and then extract attributes
+      var ctr = $doc.createElement("div");
+      ctr.innerHTML = "<div " + attributes + ">";
```

Comment:
Sounds good! I implemented a simple HTML attributes key-value parser to that end (using the same 'style' that was used for parsing knitr chunk headers).